af/shovel/centrifugation.py

```diff
@@ -1273,6 +1273,46 @@ def pop(datum):
         for key in known_keys:
             t.pop(key)
 
+class TelegramFeeder(BaseFeeder):
+    min_compat_code_ver = 6
+    data_table = sink_table = 'telegram'
+    columns = ('msm_no', 'telegram_web_failure', 'telegram_web_blocking',
+               'telegram_http_blocking', 'telegram_tcp_blocking')
+
+    @staticmethod
+    def row(msm_no, datum):
+        if datum['test_name'] != 'telegram':
+            return ''
+        t = datum['test_keys']
+
+        web_failure = t.get('telegram_web_failure', None)
+        web_status = t.get('telegram_web_status', None)
+        web_blocking = None
+        if web_status == 'ok':
+            web_blocking = False
+        elif web_status == 'blocked':
+            web_blocking = True
+
+        http_blocking = t.get('telegram_http_blocking', None)
+        tcp_blocking = t.get('telegram_tcp_blocking', None)
+        return '{:d}\t{}\t{}\t{}\t{}\n'.format(
```

Comment:
I had made a typo here in the number of columns to write into the DB, yet no exception was thrown.

@darkk is there reason to swallow these kinds of exceptions? Should maybe an issue be opened to consider at least logging them?

Reply:
From slack:
```
Yes, garbage-in-garbage-out.
That's done that way so the malformed measurement is still indexed and marked with `exc` and does not block the pipeline.
You may enable `FLAG_FAIL_FAST` to avoid exceptions suppression.
```

Reply:
More context from slack:
```
Storing exception as 32bit hashsum may sound like a strange idea, that's done to prevent database bloat in a pathological case as 1M of traces is not something trivial to deal with.
That idea is to use the hashsum to triage "failures" and pay attention to "popular" failures first.
```

Reply:
FTR, the hash is computed by [`exc_hash()`](https://github.com/ooni/pipeline/blob/065cccdfeb531e93a22d2aacc05ec05e990f99ee/af/shovel/centrifugation.py#L996-L1006).

Reply:
I documented this as part of an issue so we don't forget: https://github.com/ooni/pipeline/issues/185.